src/client.ts

```diff
@@ -131,7 +125,7 @@ export default class UnleashClient extends EventEmitter {
       return { enabled: fallback() };
     }
 
-    if (!feature || !feature.enabled) {
+    if (!feature || !this.isParentDependencySatisfied(feature, context) || !feature.enabled) {
```

Comment:
by moving parent dependency check to isFeatureEnabled we can simplify isEnabled method and resolveVariant since both of those methods call isFeatureEnabled under the hood.